### PR TITLE
Major refactoring and some fixes

### PR DESCRIPTION
**Pull request by Cryptically**
Linked with issue #65 on retaining the column ordering (at least during each session for now), considering the new UI might still take sometime to be released.

_Fix pokemon search returning wrong results due to concating of strings.
Fix select pokemon < IV
Refactored and extracted PokemonTable, PokemonTableModel
Column position persist now after calling refreshList()_

**PokemonTableModel**
- Clearly using a static number to count the columnCount was the wrong approach from the start but nobody bothered to change it.
- Same with getValueAt & getColumnName, statically coding in the return values is a pretty bad idea

**PokemonTable**
- constructNewTableModel was recreating the entire model and all the comparator on every refresh
- Thus resulting in the column reordering not persisting after a refresh.

**PokemonTab**
- Select pokemon < IV wasn't working due to trying to convert a string to double by type casting it
- Search was concating all the strings followed by contain() to find out if it matches. Causes unrelated results to be matched. E.g typing 'test', matches weedle and caterpie in my case.

And sorry for a single commit, couldn't find a suitable situation without any errors to commit a step by step changes.

### CODE REVIEW

**Comment by Ljaysoft** [comment]
About the new UI, if more people help migrate stuff to the new structure, it will go faster ;)  it will take forever if I do it all alone :S

**Comment by Cryptically** [comment]
Might have to read up on it first though, never tried javafx before.

**Comment by Wolfsblvt** [comment]
Great, thank you!
Will review it tomorrow (:

**Comment by Wolfsblvt** [comment]
Needs resolving now.

**Comment by Cryptically** [comment]
Resolved the issues

File: src/me/corriekay/pokegoutil/utils/Utilities.java [hunk 1]
```diff
@@ -70,8 +73,22 @@ public static void sleepRandom(int minMilliseconds, int maxMilliseconds) {
...
         if (e instanceof InvalidProtocolBufferException || "Contents of buffer are null".equals(message)) {
-            message = "Server hasn't responded in time. Seems to be busy. The action may have been successful though. (" + message + ")";
+            message = "Server hasn't responded in time. "
+                    + "Seems to be busy. "
+                    + "The action may have been successful though. (" + message + ")";
         }
         return message;
     }
+
+    public static String concatString(char delimeter, String... strings) {
```
Comment (nts64) [inline src/me/corriekay/pokegoutil/utils/Utilities.java:83]: `return Stream.of(strings).collect(Collectors.joining(String.valueOf(delimiter)));` does the same

**Comment by Cryptically** [comment]
Can this be merged asap. think any pr updating the old ui be conflicting with this.

**Comment by Ljaysoft** [comment]
Can be merged.

**Pull request merged**